uk.co.compucorp.civicrm.hrleaveandabsences/tests/phpunit/CRM/HRLeaveAndAbsences/BAO/LeaveRequestTest.php

```diff
@@ -216,6 +216,56 @@ public function testShouldReturnNullIfItCantFindAPublicHolidayLeaveRequestForACo
     $this->assertNull(LeaveRequest::findPublicHolidayLeaveRequest(3, $publicHoliday));
   }
 
+  public function testCanFindAPublicHolidayLeaveRequestReturnsResultWhenNonDeletedIsFalseAndPublicHolidayRequestIsDeleted() {
+    AbsencePeriodFabricator::fabricate([
+      'start_date' => CRM_Utils_Date::processDate('2017-01-01'),
+      'end_date' => CRM_Utils_Date::processDate('2017-12-01')
+    ]);
+
+    $contactID = 2;
+
+    $publicHoliday = new PublicHoliday();
+    $publicHoliday->date = '2017-01-01';
+
+    $publicHolidayRequest = PublicHolidayLeaveRequestFabricator::fabricate($contactID, $publicHoliday);
+    LeaveRequest::softDelete($publicHolidayRequest->id);
+
+    $this->assertNotNull(LeaveRequest::findPublicHolidayLeaveRequest($contactID, $publicHoliday, FALSE));
```

Comment:
Wouldn't it be better to check for the leave request id here? Just checking that it's not null is a bit too vague